python/lsst/ts/wep/estimation/aiDonut.py

```diff
@@ -0,0 +1,239 @@
+        zkStartI1 : np.ndarray
+            Starting Zernikes for I1 (unused; exists for compatibility).
+        zkStartI2 : np.ndarray or None
+            Starting Zernikes for I2 (unused; exists for compatibility).
+        nollIndices : np.ndarray
+            Noll indices for which you wish to estimate Zernike coefficients.
+        instrument : Instrument
+            Instrument object (unused; exists for compatibility).
+        saveHistory : bool
+            Whether to save the algorithm history in the self.history
+            attribute. If True, then self.history contains information
+            about the most recent time the algorithm was run.
+
+        Returns
+        -------
+        np.ndarray
+            Zernike coefficients for the provided Noll indices, estimated from
+            the images, in meters.
+        dict
+            Empty dictionary (exists for compatibility).
+        """
+        # First, let's make sure we haven't requested any Zernikes
+        # the model doesn't predict
+        if not set(nollIndices).issubset(set(self.model.nollIndices.tolist())):
+            raise ValueError(
+                f"Requested Noll indices {nollIndices.tolist()} are not "
+                "supported by the model, which can only predict some subset of "
+                f"{self.model.nollIndices.tolist()}."
+            )
+
+        # Stack inputs into a batch
+        imgs = [I1.image]
+        fxs, fys = [I1.fieldAngle[0]], [I1.fieldAngle[1]]
+        focalFlags = [1 if I1.defocalType == "intra" else 0]
+        bands = ["ugrizy".index(I1.bandLabel.value)]
+        if I2 is not None:
+            imgs.append(I2.image)
+            fxs.append(I2.fieldAngle[0])
+            fys.append(I2.fieldAngle[1])
+            focalFlags.append(1 if I2.defocalType == "intra" else 0)
+            bands.append("ugrizy".index(I2.bandLabel.value))
+
+        # Stack arrays
+        imgs_np = np.stack(imgs, axis=0)
+        fxs_np = np.array(fxs).reshape(-1, 1)
+        fys_np = np.array(fys).reshape(-1, 1)
+        focalFlags_np = np.array(focalFlags).reshape(-1, 1)
+        bands_np = np.array(bands).reshape(-1, 1)
+
+        # Convert to torch
+        imgs_tch = torch.from_numpy(imgs_np).float().to(self.device)
+        fxs_tch = torch.from_numpy(fxs_np).float().to(self.device)
+        fys_tch = torch.from_numpy(fys_np).float().to(self.device)
+        focalFlags_tch = torch.from_numpy(focalFlags_np).float().to(self.device)
+        bands_tch = torch.from_numpy(bands_np).float().to(self.device)
+
+        # Run model
+        with torch.no_grad():
+            outputs = self.model(
+                imgs_tch,
+                fxs_tch,
+                fys_tch,
+                focalFlags_tch,
+                bands_tch,
+            )
+        outputs = outputs.cpu().numpy()  # shape: (2, n_zernikes)
+
+        # Average the two estimates
+        zk = outputs.mean(axis=0)
+
+        # Only return requested nollIndices
```

Comment:
Given the if statement you have at the beginning of this method, why are these two lines needed?

Reply:
The if statement only checks that the requested Noll indices are a subset of the allowed Noll indices. These lines actually take care of retrieving the requested Noll indices